### ISSUE TITLE
[Dashboard] Workspaces

### ISSUE THREAD

**Issue by gtsiolis** [gitpod-io/gitpod#3426]
### Problem to solve

Currently, the existing overall design of the dashboard makes it hard to parse and follow information as well as association between entities related to workspaces, projects, branches, and more.

This is part of https://github.com/gitpod-io/gitpod/issues/3300. 🍕

### User Experience goal

The user should be able to navigate efficiently, have a clear view of the object attributes, relationships, and activities as well as feel confident about the state of the application and any transactional or non-transactional actions.

### Proposal

In this MVC (minimum viable change) iteration we can do the following:
1. Redesign the workspaces list.
2. Hide the object actions in a dropdown menu.
3. Show a tooltip for creation date.
4. Show a tooltip for error messages.
5. Group pending changes in a dropdown menu.
6. Introduce some help links to the documentation on the header section.
7. Show _Recent_ workspaces and filter out the rest of them (see _Status_).
8. Keep the page size option.
9. Offer a light theme.
10. Show the _Admin_ menu for admin users.

Components involved:
1. Button
2. Modal
3. Dropdown
4. Text Field
5. Item
6. Empty State
7. ...

Future steps
1. In the future, we can introduce a dark theme, and more.

### Designs

| List | List (Spacing) |
|-|-|
| ![image](https://user-images.githubusercontent.com/120486/110856041-cd520100-82bf-11eb-872c-fbd2646a29f7.png) | ![image](https://user-images.githubusercontent.com/120486/110866859-76075d00-82ce-11eb-9d2c-d37319389733.png) |

| Alert | Empty List (Has Workspaces) | Empty List (No Workspaces)
|-|-|-|
| ![image](https://user-images.githubusercontent.com/120486/111039005-7d487b00-8434-11eb-8d5f-4ff81b83e11f.png) | ![image](https://user-images.githubusercontent.com/120486/110859895-cda0cb00-82c4-11eb-9c03-4fc6941ec57e.png)  | ![image](https://user-images.githubusercontent.com/120486/110856400-43566800-82c0-11eb-9a25-50ba8c299c1f.png) |

| Select Template | Select Template (Spacing) |
|-|-|
| ![image](https://user-images.githubusercontent.com/120486/110859934-d7c2c980-82c4-11eb-9ad6-06654eebbfb3.png) | ![image](https://user-images.githubusercontent.com/120486/110859937-d8f3f680-82c4-11eb-8cba-82a3fd983bb8.png) |

See also [design specs](https://www.figma.com/file/0saJp50UikQJMOB36txeZi/Workspaces-3426).

### Documentation

No documentation changes are needed for this first iteration.

**Comment by svenefftinge** [comment]
Merged a first pass.
Todos:
 - implement search
 - update pinned and shareable
 - show ordering
 - implement pagination (more button on bottom)
 - remove status: prefix from filter
 - rendering of moment.js (seems to be off sometimes)

**Comment by svenefftinge** [comment]
More Todos:
 - [ ] Implement Empty Workspace List

**Comment by svenefftinge** [comment]
- [ ] Show GC warning
